docs/async/async-in-depth.md

```diff
@@ -0,0 +1,151 @@
+
+Assume both servers recieve 6 concurrent requests. Each request performs an I/O operation.  The server *without* async code has to queue up the 6th request until one of the 5 threads have finished the I/O-bound work and written a response. At the point that the 20th request comes in, the server might start to slow down, because the queue is getting too long.
+
+![](../../images/async/webserver-1.jpg)
+(caption) Web server with no async.
+
+As you can see, server threads can't service any other requests until the I/O they're assigned to start is finished.  That's not an ideal scenario!
+
+The server *with* async code running on it still queues up the 6th request, but because it uses `async` and `await`, each of its threads are freed up when the I/O-bound work starts, rather than when it finishes.  By the time the 20th request comes in, the queue for incoming requests will be far smaller (if it has anything in it at all), and the server won't slow down.
+
+![](../../images/async/webserver-2.jpg)
+(caption) Web server with async.
+
+As you can see, server threads are free to service more requests once they've started I/O.  The result of I/O gets queued up and the next available thread will handle writing the response for the result of the I/O.
+
+Although this is a contrived example, it works in a very similar fashion in the real world.  In fact, you can expect a server to be able to handle an order of magnitude more requests using `async` and `await` than if it were dedicating a thread for each request it receives.
+
+### What does this mean for client scenario?
+
+The biggest gain for using `async` and `await` for a client app is an increase in responsiveness.  Although you can make an app responsive by spawning threads manually, the act of spawning a thread is an expensive operation relative to just using `async` and `await`.  Especially for something like a mobile game, impacting the UI thread as little as possible where I/O is concerned is crucial.
+
+More importantly, because I/O-bound work spends virtually no time on the CPU, dedicating an entire CPU thread to perform barely any useful work would be a poor use of resources.
+
+![](../../images/async/ui.jpg)
+(caption) A UI making an async call that doesn't block the UI thread.
+
+Additionally, dispatching work to the UI thread (such as updating a UI) is very simple with `async` methods, and does not require extra work (such as calling a thread-safe delegate).
+
+## Deeper Dive into Task and Task<T> for a CPU-Bound Operation
+
+CPU-bound `async` code is a bit different than I/O-bound `async` code.  Because the work is done on the CPU, there's no way to get around dedicating a thread to the computation.  The use of `async` and `await` provides you with a clean way to interact with a background thread and keep the caller of the async method responsive.
```

Comment:
Is it worth mentioning here that 'async' and 'await' do not provide any protection around shared data. Programmers still need to worry about lock() and any other techniques to avoid concurrency issues.